sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -141,13 +140,6 @@ object DecimalType extends AbstractDataType {
     case DoubleType => DoubleDecimal
   }
 
-  private[sql] def fromLiteral(literal: Literal): DecimalType = literal.value match {
```

Comment:
It seems no need to move this one.

Reply:
See above comment

Reply:
We need to move Catalyst references from Datatype so that we can move such types into `sql/api`.